trl/trainer/sft_config.py

```diff
@@ -0,0 +1,62 @@
+# Copyright 2023 The HuggingFace Team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from dataclasses import dataclass
+from typing import Dict, Optional
+
+from transformers import TrainingArguments
+
+
+@dataclass
+class SFTConfig(TrainingArguments):
+    r"""
+    dataset_text_field (`Optional[str]`):
+        The name of the text field of the dataset, in case this is passed by a user, the trainer will automatically create a
+        `ConstantLengthDataset` based on the `dataset_text_field` argument. Defaults to None.
+    packing (`Optional[bool]`):
+            Used only in case `dataset_text_field` is passed. This argument is used by the `ConstantLengthDataset` to pack the sequences
+            of the dataset. Defaults to False.
+    max_seq_length (`Optional[int]`):
+        The maximum sequence length to use for the `ConstantLengthDataset` and for automatically creating the Dataset. Defaults to min of the smaller of the `tokenizer.model_max_length` and `1024`.
+    dataset_num_proc (`Optional[int]`):
+            The number of workers to use to tokenize the data. Only used when `packing=False`. Defaults to None.
+    dataset_batch_size (`int`):
+        The number of examples to tokenize per batch. If batch_size <= 0 or batch_size == None,
+        tokenize the full dataset as a single batch. Defaults to 1000.
+    neftune_noise_alpha (`Optional[float]`):
```

Comment:
Can this be removed now that it's part of the base `TrainingArguments` @younesbelkada ?

Reply:
Yeah totally ! we need a version check though on transformers so I think it's better to keep it ..